CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+# CoatRack Contributor Guidelines
+
+
+
+## Introduction
+
+#### Possibilities to Contribute
+
+Hello, dear prospective contributor. You can support CoatRack in many ways:
+
+- Report bugs using CoatRack's GitHub issue tracker.
+- Discuss suggested code changes in open Pull Requests.
+- Submit a bugfix.
+- Propose new features and improvement ideas.
+- Improve the documentation.
+- Clean up code.
+- Add tests.
+- Help us to fulfill the [FIWARE requirements](https://fiware-requirements.readthedocs.io/en/latest/).
+
+
+
+Note: If you want to contribute to CoatRack you need to sign the corresponding Contributors License Agreement (CLA) beforehand. Signing instructions can be found in In the corresponding section of this document.
+
+
+
+## Conventions
+
+#### Applying Best Practices
+
+The establishment of best practices in CoatRack and compliance of these is important to us. We put great effort to our review process to steadily improve the quality of our code base and we expect the same willingness from you. Here are a few practices that should always be considered: 
+
+* Code is written for readers and should be quickly understood by them. Use descriptive names, a clear architecture, short functions and small, cohesive classes. Add comments and Java Doc if required by readers to understand the code.
+* Use a consistent coding style that goes along with the already existing code. Please do not change the formatting of existing code that is not relevant to the issue you are working on.
+* Spaces should be used for indentation, not tabs.
+* Newly added code should be delivered with automated tests, for example unit tests or integration tests. Ensure to have a high code coverage.
+
+
+
+#### Workflow
+
+We use GitHub to host code, track issues and feature requests, as well as accept pull requests. We use the following workflow:
+
+1. Fork the repo. Make sure to disable the GitHub action that tries to push the code of the master branch to our 'dev' deployment environment. Of course, this step must be performed only once. All other steps are run again for each issue.
+
+2. You can either work on an existing issue or you have to create a descriptive issue before you can make code changes. 
+
+3. Create a branch from `master` for the issue you are working on. 
+
+   * Use this naming scheme: 
+     * `<commit-type>/#<issue-number>-<short-description>`. 
```

Comment:
Instead of "commit type" I would call it "issue type", as one branch/PR will include many commits.

Reply:
Done.